pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/embeddings/HuggingFaceRestEmbeddingService.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.pulsar.functions.transforms.embeddings;
+
+import com.fasterxml.jackson.annotation.JsonAlias;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Map;
+import lombok.Builder;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * EmbeddingsService implementation using HuggingFace REST API.
+ *
+ * <p>The model requested there should be trained for "sentence similarity" task.
+ */
+@Slf4j
+public class HuggingFaceRestEmbeddingService implements EmbeddingsService {
+
+  // https://huggingface.co/docs/api-inference/detailed_parameters#feature-extraction-task
+  @Data
+  @Builder
+  public static class HuggingRestConfig {
+    @Builder.Default public String hfUrl = HF_URL;
+
+    @Builder.Default Map<String, Object> options = Map.of("wait_for_model", true);
+  }
+
+  private static final String HF_URL =
+      "https://api-inference.huggingface.co/pipeline/feature-extraction/";
+  private static final ObjectMapper om = EmbeddingsService.createObjectMapper();
+
+  private final HuggingRestConfig conf;
+  private final String model;
+  private final String token;
+
+  private final URL modelUrl;
+
+  @Data
+  @Builder
+  public static class HuggingPojo {
+    @JsonAlias("inputs")
+    public List<String> inputs;
+
+    @JsonAlias("options")
+    public Map<String, Object> options;
+  }
+
+  public HuggingFaceRestEmbeddingService(String token, String model) throws MalformedURLException {
+    this(HuggingRestConfig.builder().build(), token, model);
+  }
+
+  public HuggingFaceRestEmbeddingService(HuggingRestConfig conf, String token, String model)
+      throws MalformedURLException {
+    this.conf = conf;
+    this.model = model;
+    this.token = token;
+    this.modelUrl = new URL(conf.hfUrl + model);
+
+    // TODO: try checking if model is valid https://huggingface.co/docs/datasets-server/valid
```

Comment:
we should probably have initially ad hard coded whilelist of allowed models, this way we won't let users download anything without control.